### PR TITLE
Fix "Graph2d#addCustomTime doesn't work" (#472)

### LINKED ISSUE #472: [Graph2d] addCustomTime() is not working
example: https://codepen.io/wk6/pen/zYvaRKR

When `addCustomeTime()` called by `Graph2d` object, raise error.
graph2d-addmarker-error

In 7.1.3 `vis-timeline`, `addCustomTime()` works fine (but 7.2.0 or later not works).
https://codepen.io/wk6/pen/mdeKXjy

### PR DESCRIPTION
**Pull request by ryamaguchi0220**
Close #472 

The Timeline has the `itemSet` property  but Graph2d doesn't, so the access to `itemSet.options` causes the error. 

demo: https://jsfiddle.net/1u8e4gpc/

### CODE REVIEW

**Comment by ryamaguchi0220** [comment]
I've realized that there is still one more bug that occurs when dragging markers, so I'm working on fixing it.
スクリーンショット 2020-05-27 13 10 29

**Comment by ryamaguchi0220** [comment]
The cause is `getScale()` and `getStep()` are undefined, so I added them to Graph2d a268146

**Comment by ryamaguchi0220** [comment]
@yotamberk Can I have your moment to get review?

File: lib/timeline/Core.js [hunk 1]
```diff
@@ -670,7 +670,7 @@ class Core {
     const customTime = new CustomTime(this.body, util.extend({}, this.options, {
       time : timestamp,
       id,
-      snap: this.itemSet.options.snap
+      snap: this.options.snap
```
Comment (ryamaguchi0220) [inline lib/timeline/Core.js:673]: Oops, this change makes the snapping functionality broken.

Comment (ryamaguchi0220) [reply lib/timeline/Core.js:673]: fixed: a57d0a6

Comment (yotamberk) [reply lib/timeline/Core.js:673]: I'm not sure this is still good.
The default behavior should use `this.itemSet.options.snap` so I think it is important to add that if `this.options.snap` doesn't exist, it should use `this.itemSet.options.snap`.
In the case of graph2D, a different solution should be given.

Comment (ryamaguchi0220) [reply lib/timeline/Core.js:673]: Thank you for the review and sorry for the delay.

> In the case of graph2D, a different solution should be given.

I have two questions

* Could you share why you think a different solution should be given in the case of Graph2D with me?
* Do you have any good solutions?

Comment (yotamberk) [reply lib/timeline/Core.js:673]: I would simply do something like:
``` javascript
snap: this.itemSet ? this.itemSet.options.snap : this.options.snap
```

Comment (ryamaguchi0220) [reply lib/timeline/Core.js]: I see.
Applied your suggestion: 157d0a6

**Pull request merged**